cli/tools/upgrade.rs

```diff
@@ -803,4 +823,23 @@ mod test {
     // propagation might be delated) we should not prompt
     assert_eq!(checker.should_prompt(), None);
   }
+
+  #[tokio::test]
+  async fn test_should_not_prompt_if_current_cli_version_have_changed() {
+    let env = TestUpdateCheckerEnvironment::new();
+    let file_content = CheckVersionFile {
+      last_prompt: env
+        .current_time()
+        .sub(chrono::Duration::hours(UPGRADE_CHECK_INTERVAL + 1)),
+      last_checked: env.current_time(),
+      latest_version: "1.26.2".to_string(),
+      current_version: "1.25.0".to_string(),
+    }
+    .serialize();
+    env.write_check_file(&file_content);
+    // simulate a user update
+    env.set_current_version("1.26.2");
```

Comment:
```suggestion
    env.set_current_version("61fbfabe440f1cfffa7b8d17426ffdece4d430d0");
```

I think this test case wouldn't trigger the issue because the current version would match the latest version. Let's switch to a canary version instead.